lightning/src/ln/channelmanager.rs

```diff
@@ -796,6 +808,13 @@ pub struct ChannelManager<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 	/// See `ChannelManager` struct-level documentation for lock order requirements.
 	claimable_htlcs: Mutex<HashMap<PaymentHash, (events::PaymentPurpose, Vec<ClaimableHTLC>)>>,
 
+	/// Map from payment hash to the payment data for HTLCs which we have begun claiming, but which
+	/// are waiting on a [`ChannelMonitorUpdate`] to complete in order to be surfaced to the user
+	/// as an [`events::Event::PaymentClaimed`].
+	///
+	/// See `ChannelManager` struct-level documentation for lock order requirements.
+	pending_claimed_payments: Mutex<HashMap<PaymentHash, PendingClaimingPayment>>,
```

Comment:
This needs to be added to the lock_order tree, after `pending_events`. As this is also held when `claimable_htlcs` is held, `claimable_htlcs` should be changed to be a separate level in "the main branch" of the tree. Given that we don't want to take the opportunity to change how we illustrate the lock orders :). 

Also: Would we like to add docs to `pending_claimed_payments` that the `claimable_htlcs` lock must also be held when inserting elements into `pending_claimed_payments`, as well as docs to `claimable_htlcs` that states that when inserting elements it's required to check that `pending_claimed_payments` doesn't include the payment hash, while also holding the `claimable_htlcs` lock? As that's what's protecting us from the possibility to get duplicate pending claim events IIUC.

Reply:
Hmmm I see your point, though IMO it still makes sense to provide some type of documentation of the lock orders, if anything as it's clearer and easier for new contributors. Though I'm increasingly more convinced that it's better to just have an ordered list as illustration of the lock orders, with an arbitrary order for the locks that do not currently have a connected lock order, instead of having different lock order branches in the illustration.